main.go

```diff
@@ -27,7 +33,23 @@ func main() {
 
 	config := ReadConfig(*configPath)
 
-	go bootstrapHealthCheck(config.HealthCheckPort)
+	opsStatus := make(chan *OpsStatus, 10)
+	defer close(opsStatus)
+
+	duration, err := config.getInterval()
+	if err != nil {
+		log.WithError(err).Error("Failed to parse interval")
+		return
+	}
+
+	server := buildHealthServer(opsStatus, duration, config.HealthCheckPort)
+	serverExit := make(chan os.Signal, 1)
+	defer close(serverExit)
+	mainExit := make(chan os.Signal, 1)
+	defer close(mainExit)
+	signal.Notify(serverExit, syscall.SIGINT, syscall.SIGTERM)
+	signal.Notify(mainExit, syscall.SIGINT, syscall.SIGTERM)
+	go bootstrapHealthCheck(server, serverExit)
```

Comment:
just 5 cents from my side. I don't know what is the best practice here but I`m not sure if you need 2 goroutines in main. From what I saw in other projects usually http server is initiated in the bottom line of main method without goroutine  As it has build-in goroutine, and main reacts on signals anyways, you will get what you're trying to archive out of the box without implementing this.